calico_node/startup.go

```diff
@@ -14,59 +14,657 @@
+			errored = true
+		}
+	}
+
+	if errored {
+		terminate()
+	}
+}
+
+func ensureDefaultConfig(c *client.Client, node *api.Node) error {
+	// By default we set the global reporting interval to 0 - this is
+	// different from the defaults defined in Felix.
+	//
+	// Logging to file is disabled in the felix.cfg config file.  This
+	// should always be disabled for calico/node.  By default we log to
+	// screen - set the default logging value that we desire.
+	if err := ensureGlobalFelixConfig(c, "ReportingIntervalSecs", "0"); err != nil {
+		return err
+	} else if err = ensureGlobalFelixConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	}
+
+	// Set the default values for some of the global BGP config values and
+	// per-node directory structure.
+	// These are required by both confd and the GoBGP daemon.  Some of this
+	// can only be done directly by the backend (since it requires access to
+	// datastore features not exposed in the main API).
+	//
+	// TODO: This is only required for the current BIRD and GoBGP integrations,
+	//       but should be removed once we switch over to a better watcher interface.
+	if err := ensureGlobalBGPConfig(c, "node_mesh", fmt.Sprintf("{\"enabled\": %v}", client.GlobalDefaultNodeToNodeMesh)); err != nil {
+		return err
+	} else if err := ensureGlobalBGPConfig(c, "as_num", strconv.Itoa(client.GlobalDefaultASNumber)); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "loglevel", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = c.Backend.EnsureCalicoNodeInitialized(node.Metadata.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures that the supplied global felix config value
+// is initialized, and if not initialize it with the supplied default.
+func ensureGlobalFelixConfig(c *client.Client, key, def string) error {
+	if val, assigned, err := c.Config().GetFelixConfig(key, ""); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, "", def)
+	} else {
+		log.Infof("Global Felix value already assigned %s=%s", key, val)
+		return nil
+	}
+}
+
+// ensureGlobalBGPConfig ensures that the supplied global BGP config value
+// is initialized, and if not initialize it with the supplied default.
+func ensureGlobalBGPConfig(c *client.Client, key, def string) error {
+	if val, assigned, err := c.Config().GetBGPConfig(key, ""); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetBGPConfig(key, "", def)
+	} else {
+		log.Infof("Global BGP value already assigned %s=%s", key, val)
+		return nil
+	}
+}
+
+// message() prints a message to screen and to log.  A newline terminator is
+// not required in the format string.
+func message(format string, args ...interface{}) {
```

Comment:
I'm not so sure.  Docker logs are important since that's the main way that we do logging in Kubernetes when using the self-hosted approach, so this means one of our most used deployment mechanisms will show duplicate logs.